core/src/main/java/org/apache/accumulo/core/file/rfile/RFile.java

```diff
@@ -1166,29 +1165,27 @@ public Reader(CachableBlockFile.Reader rdr) throws IOException {
           throw new IOException("Did not see expected version, saw " + ver);
 
         int size = mb.readInt();
-        currentReaders = new LocalityGroupReader[size];
+        currentReaders = new ArrayList<>(size);
 
-        deepCopies = new LinkedList<>();
+        deepCopies = new ArrayList<>();
```

Comment:
We only ever do `add()` on this, which can cause problems on certain system patterns which use a lot of deepCopies, if it results in array resizing. `LinkedList` is probably better here for those use cases, and not worse for other use cases. I'd leave this one the way it was.